tests/integration/monitoring/test_monitoring.py

```diff
@@ -152,3 +153,94 @@ def test_disable_monitoring_on_gatway_only(port_generator, executor):
+
+
+def test_requests_size(port_generator, executor):
+    port0 = port_generator()
+    port1 = port_generator()
+
+    with Flow(monitoring=True, port_monitoring=port0).add(
+        uses=executor, port_monitoring=port1
+    ) as f:
+
+        f.post('/foo', inputs=DocumentArray.empty(size=1))
+
+        resp = req.get(f'http://localhost:{port1}/')  # enable on port0
+        assert resp.status_code == 200
+
+        assert (
+            f'jina_request_size_bytes_count{{executor="DummyExecutor",executor_endpoint="/foo",runtime_name="executor0/rep-0"}} 1.0'
+            in str(resp.content)
+        )
+
+        def _get_request_bytes_size():
+            resp = req.get(f'http://localhost:{port1}/')  # enable on port0
+
+            resp_lines = str(resp.content).split('\\n')
+            byte_line = [
+                line
+                for line in resp_lines
+                if 'jina_request_size_bytes_sum{executor="DummyExecutor"' in line
+            ]
+
+            return float(byte_line[0][-5:])
+
+        measured_request_bytes_sum_init = _get_request_bytes_size()
+        f.post('/foo', inputs=DocumentArray.empty(size=1))
+        measured_request_bytes_sum = _get_request_bytes_size()
+
+        assert measured_request_bytes_sum > measured_request_bytes_sum_init
+
+
+def test_pending_request(port_generator):
+    port0 = port_generator()
+    port1 = port_generator()
+
+    class SlowExecutor(Executor):
+        @requests
+        def foo(self, docs, **kwargs):
+            time.sleep(5)
+
+    with Flow(monitoring=True, port_monitoring=port0).add(
+        uses=SlowExecutor, port_monitoring=port1
+    ) as f:
+
+        def _send_request():
+            f.search(inputs=DocumentArray.empty(size=1))
+
+        def _assert_pending_value(val: str):
+            resp = req.get(f'http://localhost:{port0}/')
+            assert resp.status_code == 200
+            assert (
+                f'jina_number_of_pending_requests{{runtime_name="gateway/rep-0/GRPCGatewayRuntime"}} {val}'
+                in str(resp.content)
+            )
+
+        _assert_while = lambda: _assert_pending_value(
+            '1.0'
+        )  # while the request is being processed the counter is at one
+        _assert_after = lambda: _assert_pending_value(
+            '0.0'
+        )  # but before and after it is at 0
+        _assert_before = lambda: _assert_pending_value(
+            '0.0'
+        )  # but before and after it is at 0
+
+        p_send = multiprocessing.Process(target=_send_request)
+        p_before = multiprocessing.Process(target=_assert_before)
+        p_while = multiprocessing.Process(target=_assert_while)
+
+        p_before.start()
+        time.sleep(1)
+        p_send.start()
+        time.sleep(1)
+        p_while.start()
+
+        for p in [p_before, p_send, p_while]:
+            p.join()
+
+        for p in [p_before, p_send, p_while]:
+            p.terminate()
+            assert not p.exitcode
```

Comment:
collect the exit codes and assert after all of them have been terminated, to avoid timeouts